pkg/client/daemon/outbound.go

```diff
@@ -184,9 +209,8 @@ func (o *outbound) resolveInCluster(c context.Context, qType uint16, query strin
 		}
 	}
 
-	// Give the cluster lookup a reasonable timeout
-	tos := client.GetConfig(c).Timeouts
-	c, cancel := tos.TimeoutContext(c, client.TimeoutTrafficManagerAPI)
+	// Give the cluster lookup a reasonable timeout.
+	c, cancel := context.WithTimeout(c, 4*time.Second)
```

Comment:
What's the rationale for making this timeout hard-coded to 4 seconds? Is it just so we don't get hung if the cluster lookup fails?

Reply:
Yes, I think we want a shorter timeout here than the default that is used in resolv.conf (5 seconds). We should make this timeout configurable in the future, but I didn't want to start yet another cycle of release notes and docs just for that.